src/main/scala/com/fulcrumgenomics/vcf/MakeMixtureVcf.scala

```diff
@@ -0,0 +1,268 @@
+    val header = new VCFHeader(Collections.emptySet[VCFHeaderLine](), util.Arrays.asList(samples:_*))
+
+    in.getFilterLines.foreach(header.addMetaDataLine)
+    in.getFormatHeaderLines.foreach(header.addMetaDataLine)
+    in.getInfoHeaderLines.foreach(header.addMetaDataLine)
+    lines.foreach(header.addMetaDataLine)
+
+    header.setSequenceDictionary(in.getSequenceDictionary)
+
+    val writer = new VariantContextWriterBuilder()
+      .setOutputFile(output.toFile)
+      .setOption(Options.INDEX_ON_THE_FLY)
+      .setReferenceDictionary(in.getSequenceDictionary)
+      .setOption(Options.ALLOW_MISSING_FIELDS_IN_HEADER)
+      .build()
+
+    writer.writeHeader(header)
+    writer
+  }
+}
+
+
+@clp(group=ClpGroups.VcfOrBcf, description=
+  """Creates a VCF with one sample whose genotypes are a mixture of other samples'.
+    |
+    |The input VCF must contain all samples to be mixed, and may optionally contain other samples.
+    |Sample mixtures can be specified in one of several ways:
+    |  1. --samples s1 s2 s3: specifies that the samples with names s1, s2 and s3 should be mixed equally.
+    |  2. --samples s1@0.1 s2@0.1 s3@0.8: specifies that the three samples should be mixed at 0.1, 0.1 and 0.8
+    |  3. --samples s1@0.1 s2 s3: specifies that s1 should form 0.1 of the mixture and that the remaining (0.9)
+    |                             should be split amongst s2 and s3
+    |  4. If no sample names are given, all samples in the input VCF will be used, mixing equally
+    |
+    |The input samples are assumed to be diploid with the allele fraction defined by the genotype (0/0, 0/1 or 1/1).
+    |The allele-fraction-field option may be specified, in which case the allele fraction of the input genotypes
+    |will be retrieved from the specified FORMAT field in the VCF genotypes.  All genotypes except hom-ref and
+    |no-call genotypes must have this field present if supplied.
+    |
+    |If the no-call-is-hom-ref flag is true (the default) then no-call genotypes in the input VCF are interpreted
+    |as hom-ref genotypes.  If it is false, any location with a no-call genotype will be emitted as a no-call in
+    |the mixture, and will be filtered.
+  """)
+class MakeMixtureVcf
+( @arg(flag="i", doc="Input VCF containing genotypes for the samples to be mixed.") val input: PathToVcf,
+  @arg(flag="o", doc="Output VCF of mixture sample.") val output: PathToVcf,
+  @arg(flag="s", doc="Samples to mix. See general usage for format and examples.", minElements=0) val samples: Seq[String] = Seq.empty,
+  @arg(flag="S", doc="Output sample name.") val outputSampleName: String,
+  @arg(flag="N", doc="Treat no-calls for samples as hom-ref genotypes.") val noCallIsHomRef: Boolean = true,
+  @arg(flag="a", doc="Format field containing allele fraction.") val alleleFractionField: Option[String] = None,
+  @arg(flag="p", doc="Digits of precision in generated allele fractions.") val precision: Int = 5
```

Comment:
Why is six always enough?  I think, especially in cases where you have awkward numbers of samples (e.g. 3) and can generate proportions with recurring decimals, it's important to have enough information but not _too_ much.  That might differ and it's easy enough to expose it as a parameter with a default.